clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -360,6 +362,30 @@ private PushJobSetting getPushJobSetting(VeniceProperties props) {
     pushJobSettingToReturn.suppressEndOfPushMessage = props.getBoolean(SUPPRESS_END_OF_PUSH_MESSAGE, false);
     pushJobSettingToReturn.deferVersionSwap = props.getBoolean(DEFER_VERSION_SWAP, false);
     pushJobSettingToReturn.repushTTLEnabled = props.getBoolean(REPUSH_TTL_ENABLE, false);
+
+    if (pushJobSettingToReturn.repushTTLEnabled && !pushJobSettingToReturn.isSourceKafka) {
+      throw new VeniceException("Repush with TTL is only supported while using Kafka Input Format");
+    }
+
+    pushJobSettingToReturn.repushTTLStartTimeMs = -1;
+    if (pushJobSettingToReturn.repushTTLEnabled) {
+      long repushTtlSeconds = props.getLong(REPUSH_TTL_SECONDS, -1);
+      long repushTtlStartTimestamp = props.getLong(REPUSH_TTL_START_TIMESTAMP, -1);
+
+      if (repushTtlSeconds >= 0 && repushTtlStartTimestamp >= 0) {
+        String message =
+            "Both " + REPUSH_TTL_SECONDS + " and " + REPUSH_TTL_START_TIMESTAMP + " are set. Please set only one.";
+        throw new VeniceException(message);
+      }
+
+      if (repushTtlSeconds >= 0) {
+        pushJobSettingToReturn.repushTTLStartTimeMs =
+            pushJobSettingToReturn.jobStartTimeMs - (repushTtlSeconds * Time.MS_PER_SECOND);
+      } else if (repushTtlStartTimestamp >= 0) {
```

Comment:
No `else if` needed, right?  may be a ternary op:

`pushJobSettingToReturn.repushTTLStartTimeMs = repushTtlStartTimestamp >= 0 ? repushTtlStartTimestamp : pushJobSettingToReturn.jobStartTimeMs - (repushTtlSeconds * Time.MS_PER_SECOND)

Reply:
There are 3 ways to configure `repushTTLStartTimeMs` - 
1. `repush.ttl.seconds` (Optional)
2. `repush.ttl.start.timestamp` (Optional)
3. Store rewind time config

This piece of code in this thread runs during config time - and not run method (for improving testability) and checks for Points 1 and 2. In the `run` method, before setting the final job config on the compute engine, we check if `jobStartTimeMs` is still `<0` (unset), then we use the one from hybrid configs.